File: apps/mobile/ios/Visvine/Repositories/EventsRepository.swift
import Foundation

struct EventsRepository {
    private let api = APIClient.shared

    func getEvents(spaceId: String) async -> APIResult<[Event]> {
        let res: APIResult<EventsResponse> = await api.request(
            "/api/events", query: ["spaceId": spaceId]
        )
        switch res {
        case .success(let r): return .success(r.events)
        case .failure(let m): return .failure(m)
        }
    }

    func getEvent(eventId: String) async -> APIResult<Event> {
        await api.request("/api/events/\(eventId)")
    }
}

File: apps/mobile/android/app/src/main/java/com/visvine/mobile/data/repository/EventsRepository.kt
package com.visvine.mobile.data.repository

import com.visvine.mobile.data.model.Event
import com.visvine.mobile.data.remote.ApiResult
import com.visvine.mobile.data.remote.VisvineApi
import com.visvine.mobile.data.remote.safeApiCall
import kotlinx.serialization.json.Json
import javax.inject.Inject
import javax.inject.Singleton

@Singleton
class EventsRepository @Inject constructor(
    private val api: VisvineApi,
    private val json: Json,
) {
    suspend fun getEvents(spaceId: String): ApiResult<List<Event>> =
        when (val res = safeApiCall(json) { api.getEvents(spaceId) }) {
            is ApiResult.Success -> ApiResult.Success(res.data.events)
            is ApiResult.Failure -> res
        }

    suspend fun getEvent(eventId: String): ApiResult<Event> =
        safeApiCall(json) { api.getEvent(eventId) }
}
